src/main/java/net/dries007/tfc/common/blockentities/NestBoxBlockEntity.java

```diff
@@ -42,17 +46,26 @@ public static void serverTick(Level level, BlockPos pos, BlockState state, NestB
             Entity sitter = Seat.getSittingEntity(level, pos);
             if (sitter instanceof OviparousAnimal bird)
             {
-                if (bird.isReadyForAnimalProduct())
+                if (bird.isReadyForAnimalProduct() && !nest.isFull())
                 {
                     if (bird.getRandom().nextInt(7) == 0)
                     {
-                        Helpers.playSound(level, pos, SoundEvents.CHICKEN_EGG);
-                        if (Helpers.insertOne(level, pos, TFCBlockEntities.NEST_BOX.get(), bird.makeEgg()))
+                        var event = bird.makeEggEvent();
+
+                        if (!MinecraftForge.EVENT_BUS.post(event))
                         {
-                            bird.setFertilized(false);
-                            bird.setProductsCooldown();
-                            bird.stopRiding();
-                            nest.markForSync();
+                            if (Helpers.insertOne(Optional.of(nest), event.getProduct())) {
+                                Helpers.playSound(level, pos, SoundEvents.CHICKEN_EGG);
+
+                                bird.addUses(event.getUses());
+                                bird.setFertilized(false);
+                                bird.setProductsCooldown();
+                                bird.stopRiding();
+                                nest.markForSync();
+                            } else {
+                                // the nest box is full, leave it be
+                                bird.stopRiding();
+                            }
```

Comment:
Please import and apply codestyle! We have an XML in the repo. Failing that, Please organize your braces like so:

```
if
{
..
}
else
{
...
}
```

Reply:
I was able to import the codestyle into Intellij Idea, although just blindly running a reformat on the modified code did touch some unrelated sections. I did try my best to apply it where relevant though.